### ISSUE TITLE
"index.php" shows up in canonical, while it shouldn't. 

### ISSUE THREAD

**Issue by bobdenotter** [bolt/bolt#4793]
Silex allows getting URLs with `index.php` in them, in the routing. This is fine, but the *canonical* shouldn't show it: 

![screenshot_2016-02-02_18_53_17](https://cloud.githubusercontent.com/assets/1833361/12758865/95a44622-c9de-11e5-8cfc-4c016cbd6691.png)

**Comment by GawainLynch** [comment]
Erm… you have `index.php` in the route… removing that in that circumstance would be a bad idea… what is right/wrong/allowed and how much overhead do we want to add?

**Comment by bobdenotter** [comment]
In this case, `index.php/entry/1` is literally the same as `/entry/1`. We don't need to redirect it, but the canonical _should_ show the one without it, because "duplicate content" is frowned upon by Google. 

> how much overhead do we want to add?

Not much, I hope :-)

**Comment by GawainLynch** [comment]
Google won't see that as duplicate. It simply crawls the sitemap, and links… if nothing points to `/foo/bar` it won't get index. Matt Cutts has a bunch of stuff on this… somewhere.

Problem being that we'd **have** to do **more** db queries for a route that might well be valid.

**Comment by bobdenotter** [comment]
Yes, i'm positive it does. If the URL *works* (which in this case it does) and displays the same content as another URL (which it also does in this case), Google will mark it as duplicate content which is bad for ranking. 

See: https://moz.com/learn/seo/duplicate-content

**Comment by GawainLynch** [comment]
> Yes, i'm positive it does.

I am also pretty positive it doesn't… and that is nothing more than a blog post of a company selling stuff… Just sayin' :wink:

**Comment by bobdenotter** [comment]
I have a slight feeling we're talking about something else then. The thing I am talking about is a real issue, and it has a real easy fix. I see no reason not to fix it. 

PR in 2 minutes.

**Comment by GawainLynch** [comment]
Dev meeting in less than 1… :wink:

**Comment by CarsonF** [comment]
Yeah I'm pretty sure we don't and can't assume that that canonical url would be valid

**Comment by KrystianWolf** [comment]
I have reported this issue on IRC. We're using Bolt for our project. When I went to Google Webmaster Tools I saw this:

![webmaster-tools-duplicates](https://cloud.githubusercontent.com/assets/17032049/12760720/fcbf4ccc-c9e7-11e5-9eda-59f713e501da.jpg)

We don't have any links to the "index.php" version. It just happened... ;)

**Comment by GawainLynch** [comment]
What is the site's base URL?

**Comment by KrystianWolf** [comment]
Hi Gawain, site's name is http://www.najlepszekonto.pl/

**Comment by KrystianWolf** [comment]
I've found promissing solution that uses .htaccess rules:

```
        # Removes index.php slug from URL
        RewriteCond %{THE_REQUEST} ^GET.*index\.php [NC]
        RewriteRule (.*?)index\.php/*(.*) /$1$2 [R=302,NE,L]

        # Directs all web requests through the site index.php file
        RewriteCond %{REQUEST_FILENAME} !-f
        RewriteCond %{REQUEST_FILENAME} !-d
        RewriteRule ^(.*)$ /index.php/$1 [L]
```
Done some testing on test server and it looks OK. I'm not dev though, so maybe you will find a better way or make some improvements to the code above.

**Comment by CarsonF** [comment]
@KrystianWolf I just use this:
```apache
  # Pass all requests not referring directly to files in the filesystem to index.php
  RewriteCond %{REQUEST_FILENAME} !-f
  RewriteCond %{REQUEST_FILENAME} !-d
  RewriteCond %{REQUEST_FILENAME} !-l
  RewriteRule ^ index.php [L]
```

I think it might be your second `RewriteRule` that is throwing you off.

**Comment by bobdenotter** [comment]
Fixed by #4941. Thanks, @CarsonF, @GawainLynch!

See: 

![screenshot_2016-03-02_21_22_59](https://cloud.githubusercontent.com/assets/1833361/13474209/3de52442-e0bd-11e5-87d1-95461d6dd01f.png)

**Comment by CarsonF** [comment]
Uhhhh wut. I didn't change anything regarding that...did I? :confused:
